search/indexing_api.py

```diff
@@ -0,0 +1,95 @@
+"""
+
+"""
+from django.conf import settings
+
+from search.connection import get_conn
+
+
+GLOBAL_DOC_TYPE = '_doc'
+OP_CREATE = 'create'
+OP_INDEX = 'index'
+
+COMBINED_MAPPING = {
+    'doc_type': {'type': 'keyword'},
+    'author': {'type': 'keyword'},
+    'channel_title': {'type': 'keyword'},
+    'text': {'type': 'text'},
+    'score': {'type': 'long'},
+    'created': {'type': 'date'},
+    'deleted': {'type': 'boolean'},
+    'removed': {'type': 'boolean'},
+    'post_id': {'type': 'keyword'},
+    'post_title': {'type': 'text'},
+    'post_link_url': {'type': 'text'},
+    'num_comments': {'type': 'long'},
+    'comment_id': {'type': 'keyword'},
+    'parent_comment_id': {'type': 'keyword'},
+}
+
+
+def gen_post_id(reddit_obj_id):
+    return 'p_{}'.format(reddit_obj_id)
+
+
+def _build_mapping_dict_for_request(mapping):
+    return {
+        GLOBAL_DOC_TYPE: {
+            "properties": mapping
+        }
+    }
+
+
+def clear_and_create_index(skip_mapping=False):
+    """
+    Wipe and recreate index and mapping. No indexing is done.
+
+    Args:
+        skip_mapping (bool): If true, don't set any mapping
+    """
+    index_name = settings.ELASTICSEARCH_INDEX
+    mapping = _build_mapping_dict_for_request(COMBINED_MAPPING)
+
+    conn = get_conn(verify=False)
+    if conn.indices.exists(index_name):
+        conn.indices.delete(index_name)
+    # from https://www.elastic.co/guide/en/elasticsearch/guide/current/asciifolding-token-filter.html
+    conn.indices.create(index_name, body={
+        'settings': {
+            'analysis': {
+                'analyzer': {
+                    'folding': {
+                        'type': 'custom',
+                        'tokenizer': 'standard',
+                        'filter': [
+                            'lowercase',
+                            'asciifolding',  # remove accents if we use folding analyzer
+                        ]
+                    }
+                }
+            }
+        },
+        'mappings': {
+            **({} if skip_mapping else mapping),
```

Comment:
Would suggest either simplifying this to not use `**` or otherwise use `copy.deepcopy` if a seperate copy is what we need here.